app/src/main/java/org/groundplatform/android/ui/home/mapcontainer/jobs/LoiJobSheet.kt

```diff
@@ -130,7 +130,7 @@ private fun ModalContents(
         fontSize = 16.sp,
       )
       // NOTE(#2539): The DataCollectionFragment will crash if there are no non-LOI tasks.
-      if (canUserSubmitData && loi.job.hasNonLoiTasks()) {
+      if (canUserSubmitData && loi.job.hasNonLoiTasks() && loi.isPredefined == true) {
```

Comment:
Isn't `loi.isPredefined == true` the same as jsut `loi.isPredefined`?

Reply:
by default loi.isPredefined is null hence == true is required here.